cpp/src/groupby/hash/groupby.cu

```diff
@@ -647,12 +647,12 @@ bool can_use_hash_groupby(table_view const& keys, host_span<aggregation_request
   // Currently, structs are not supported in any of hash-based aggregations.
   // Therefore, if any request contains structs then we must fallback to sort-based aggregations.
   // TODO: Support structs in hash-based aggregations.
-  auto const has_struct =
+  auto const all_no_structs =
```

Comment:
This should be a `std::any_of` for `r.values.type().id() == type_id::STRUCT`.

Reply:
Done